backend/src/main/java/org/cryptomator/hub/api/VaultResource.java

```diff
@@ -236,81 +252,109 @@ private Response removeAuthority(UUID vaultId, String authorityId) {
+	public Response legacyUnlock(@PathParam("vaultId") UUID vaultId, @PathParam("deviceId") @ValidId String deviceId) {
 		var vault = Vault.<Vault>findByIdOptional(vaultId).orElseThrow(NotFoundException::new);
 		if (vault.archived) {
 			throw new GoneException("Vault is archived.");
 		}
 
-		var usedSeats = EffectiveVaultAccess.countEffectiveVaultUsers();
+		var usedSeats = EffectiveVaultAccess.countSeatOccupyingUsers();
 		if (usedSeats > license.getAvailableSeats()) {
 			throw new PaymentRequiredException("Number of effective vault users exceeds available license seats");
 		}
 
-		var access = AccessToken.unlock(vaultId, deviceId, jwt.getSubject());
+		var access = LegacyAccessToken.unlock(vaultId, deviceId, jwt.getSubject());
 		if (access != null) {
 			UnlockVaultEvent.log(jwt.getSubject(), vaultId, deviceId, UnlockVaultEvent.Result.SUCCESS);
 			var subscriptionStateHeaderName = "Hub-Subscription-State";
 			var subscriptionStateHeaderValue = license.isSet() ? "ACTIVE" : "INACTIVE"; // license expiration is not checked here, because it is checked in the ActiveLicense filter
 			return Response.ok(access.jwe).header(subscriptionStateHeaderName, subscriptionStateHeaderValue).build();
-		} else if (Device.findById(deviceId) == null) {
-			throw new NotFoundException("No such device.");
 		} else {
 			UnlockVaultEvent.log(jwt.getSubject(), vaultId, deviceId, UnlockVaultEvent.Result.UNAUTHORIZED);
 			throw new ForbiddenException("Access to this device not granted.");
 		}
 	}
 
+	@GET
+	@Path("/{vaultId}/access-token")
+	@RolesAllowed("user")
+	@Transactional
+	@Produces(MediaType.TEXT_PLAIN)
+	@Operation(summary = "get the user-specific vault key", description = "retrieves a jwe containing the vault key, encrypted for the current user")
+	@APIResponse(responseCode = "200")
+	@APIResponse(responseCode = "402", description = "number of effective vault users exceeds available license seats")
+	@APIResponse(responseCode = "403", description = "user not authorized to access this vault")
+	@APIResponse(responseCode = "404", description = "unknown vault")
+	@APIResponse(responseCode = "410", description = "Vault is archived")
+	@ActiveLicense // may throw 402
+	public String unlock(@PathParam("vaultId") UUID vaultId) {
+		var vault = Vault.<Vault>findByIdOptional(vaultId).orElseThrow(NotFoundException::new);
+		if (vault.archived) {
+			throw new GoneException("Vault is archived.");
+		}
+
+		var usedSeats = EffectiveVaultAccess.countSeatOccupyingUsers();
+		if (usedSeats > license.getAvailableSeats()) {
+			throw new PaymentRequiredException("Number of effective vault users exceeds available license seats");
+		}
+
+		var access = AccessToken.unlock(vaultId, jwt.getSubject());
+		if (access != null) {
+			return access.vaultKey;
+		} else if (Vault.findById(vaultId) == null) {
+			throw new NotFoundException("No such vault.");
+		} else {
+			throw new ForbiddenException("Access to this vault not granted.");
+		}
```

Comment:
`device_id` is `NOT NULL`... Not exactly compatible with this refactoring. I see two options:

1. Remove this column in this PR
2. Don't add the column in the first place in 1.2.0. It won't last very long anyways 😉

Reply:
Oh... I've looked into the flowchart again. How about the following:

1. In `/api/vaults/{vaultId}/access-token`, we only add the audit event log for the last case, meaning if the vault unlock failed due to missing vault access.
2. In `/api/devices/{deviceId}`, we add the audit event log for a successful unlock. But I wonder, why did we remove the case `NO_SUCH_DEVICE` then?

I know that these APIs can also be called in reverse but you probably have this "problem" in any order. Even in this resource, the first case doesn't have to lead to a successful unlock because the user might still have to register their device.

Reply:
The latter does not need to be an unlock. There can be plenty reasons, why a client GETs device info.

Reply:
added in e9411b4